featuretools/entityset/entityset.py

```diff
@@ -526,403 +556,459 @@ def entity_from_dataframe(self,
-            if type(base_entity[v]) == vtypes.DatetimeTimeIndex:
-                transfer_types[v] = vtypes.Datetime
-            elif type(base_entity[v]) == vtypes.NumericTimeIndex:
-                transfer_types[v] = vtypes.Numeric
-            else:
-                transfer_types[v] = type(base_entity[v])
-
-        # create and add new entity
-        new_entity_df = self[base_entity_id].df.copy()
-
-        if make_time_index is None and base_entity.time_index is not None:
+        # Types will be a tuple of (logical_type, semantic_tags, column_metadata, column_description)
+        transfer_types[index] = (base_dataframe.ww.logical_types[index],
+                                 base_dataframe.ww.semantic_tags[index],
+                                 base_dataframe.ww.columns[index].metadata,
+                                 base_dataframe.ww.columns[index].description)
+        for col_name in additional_columns + copy_columns:
+            # Remove any existing time index tags
+            transfer_types[col_name] = (base_dataframe.ww.logical_types[col_name],
+                                        (base_dataframe.ww.semantic_tags[col_name] - {'time_index'}),
+                                        base_dataframe.ww.columns[col_name].metadata,
+                                        base_dataframe.ww.columns[col_name].description)
+
+        # create and add new dataframe
+        new_dataframe = self[base_dataframe_name].copy()
+
+        if make_time_index is None and base_dataframe.ww.time_index is not None:
             make_time_index = True
 
         if isinstance(make_time_index, str):
             # Set the new time index to make_time_index.
             base_time_index = make_time_index
-            new_entity_time_index = make_time_index
-            already_sorted = (new_entity_time_index == base_entity.time_index)
+            new_dataframe_time_index = make_time_index
+            already_sorted = (new_dataframe_time_index == base_dataframe.ww.time_index)
         elif make_time_index:
-            # Create a new time index based on the base entity time index.
-            base_time_index = base_entity.time_index
-            if new_entity_time_index is None:
-                new_entity_time_index = "first_%s_time" % (base_entity.id)
+            # Create a new time index based on the base dataframe time index.
+            base_time_index = base_dataframe.ww.time_index
+            if new_dataframe_time_index is None:
+                new_dataframe_time_index = "first_%s_time" % (base_dataframe.ww.name)
 
             already_sorted = True
 
-            assert base_entity.time_index is not None, \
-                "Base entity doesn't have time_index defined"
+            assert base_dataframe.ww.time_index is not None, \
+                "Base dataframe doesn't have time_index defined"
+
+            if base_time_index not in [v for v in additional_columns]:
```

Comment:
I think this should be `copy_columns`, not `additional_columns`.  on lines 690-692 we already check for the base time index being in additional columns and error if it is

Reply:
yep, that makes sense to me. 

I've been looking into the case where we do have the base dataframe's time index in `copy_columns`, and ran into an issue where we include the original time index's column name as an element in `transfer_types`, so it causes an error when initializing Woodwork, since we don't include that column name in the normalized dataframe. Added some tests to catch this case as well as a fix for it.